x/emissions/keeper/inference_synthesis/forecast_implied_inferences.go

```diff
@@ -74,7 +74,14 @@ func CalcForcastImpliedInferences(
 			// Map inferer -> forecast element => only one (latest in array) forecast element per inferer
 			forecastElementsByInferer := make(map[Worker]*emissions.ForecastElement, 0)
 			for _, el := range forecast.ForecastElements {
-				forecastElementsByInferer[el.Inferer] = el
+				for worker := range inferenceByWorker {
```

Comment:
was thinking it'd solved when the payload of inferences is committed to the chain but think you're right -- it's safer to have these checks anyway